contrib/win32/win32compat/shell-host.c

```diff
@@ -663,53 +790,41 @@ ProcessEvent(void *p)
 		wX = LOWORD(idObject);
 		wY = HIWORD(idObject);
 		
-		SMALL_RECT readRect;
 		readRect.Top = wY;
 		readRect.Bottom = wY;
 		readRect.Left = wX;
-		readRect.Right = ConSRWidth();
+		readRect.Right = wX;
 		
-		/* Set cursor location based on the reported location from the message */
-		CalculateAndSetCursor(pipe_out, wX, wY);
-		
-		COORD coordBufSize;
-		coordBufSize.Y = readRect.Bottom - readRect.Top + 1;
-		coordBufSize.X = readRect.Right - readRect.Left + 1;
+		coordBufSize.Y = 1;
+		coordBufSize.X = 1;
+
+		if (!IS_TT_VTX(termType) || wX < currentCol || wY < currentLine) {
+			/* send full line */
+			readRect.Right = ConSRWidth();
+			coordBufSize.Y += readRect.Bottom - readRect.Top;
+			coordBufSize.X += readRect.Right - readRect.Left;
+		}
+		bufferSize = coordBufSize.X * coordBufSize.Y;
+		if (bufferSize > MAX_EXPECTED_BUFFER_SIZE)
```

Comment:
This looks wrong...
#define MAX_EXPECTED_BUFFER_SIZE 1024

coordBufSize.X, coordBufSize.Y refers to the width and height of the visible window..
If width and height are the following values then we will return ERROR_INVALID_PARAMETER here?
coordBufSize.X = 237, 
coordBufSize.Y = 71, 

![image](https://cloud.githubusercontent.com/assets/23668037/26272181/574799c6-3cc7-11e7-95e9-8a2b01e043f4.png)